test/teiserver/lobby/libs/lobby_restrictions_test.exs

```diff
@@ -0,0 +1,58 @@
+defmodule Teiserver.Lobby.Libs.LobbyRestrictionsTest do
+  @moduledoc false
+  use Teiserver.ServerCase, async: false
```

Comment:
I don't think you need `ServerCase`, `ExUnit` should be enough, this way you don't need to setup the entire server.
You can also make that synchronous, since you're not even touching the DB, you're only testing pure functions.